api/src/test/resources/org/openmrs/api/include/ConceptServiceTest-conceptReferenceRange.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!--
+
+    This Source Code Form is subject to the terms of the Mozilla Public License,
+    v. 2.0. If a copy of the MPL was not distributed with this file, You can
+    obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+    the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+
+    Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+    graphic logo is a trademark of OpenMRS Inc.
+
+-->
+<dataset>
+	<concept concept_id="1" retired="false" datatype_id="3" class_id="7" is_set="false" creator="1" date_created="2005-01-01 00:00:00.0" uuid="957eba27-2b38-43e8-91a9-4dfe3956a32d"/>
+	<concept concept_id="2" retired="false" datatype_id="1" class_id="7" is_set="false" creator="1" date_created="2005-01-01 00:00:00.0" uuid="92b910bd-298c-4ecf-a632-661ae2f446bf"/>
+	<concept_description concept_description_id="1" concept_id="1" description="this is a description" locale="en" creator="1" date_created="2005-01-01 00:00:00.0" uuid="0a6f34ea-043a-4e69-b418-1d7bb66b99b6"/>
+	<concept_description concept_description_id="2" concept_id="2" description="this is another description" locale="en" creator="1" date_created="2005-01-01 00:00:00.0" uuid="a72dac37-b593-46a1-a24b-e3da6dbcc521"/>
+	<concept_name concept_id="1" name="Blood pressure" locale="en" creator="1" date_created="2005-01-01 00:00:00.0" concept_name_id="1" concept_name_type="FULLY_SPECIFIED" locale_preferred="0" voided="false" uuid="8edfd54f-b189-43dd-86e0-913591978e48"/>
+	<concept_numeric concept_id="2" hi_absolute="123.0" allow_decimal="false"/>
+	<concept_reference_range concept_reference_range_id="3" concept_id="2" criteria="${fn.getAge(1-3)}" hi_absolute="120.0" hi_normal="118.0" low_normal="80" uuid="8edfd54f-b189-43dd-86e0-913591978e48"/>
```

Comment:
In fn.getAge(), what does `1-3` mean?

Reply:
Age group (1 year to 3 years). Do you have suggestions on how we can represent the age group?

I used age group based on Grace's introduction on this conversation: https://talk.openmrs.org/t/age-sensitive-ranges/41654

Reply:
This is less flexible than what i had in mind. My vision for the criteria was to be something like: `fn.getAge() >= 1 && fn.getAge() <= 3`

Reply:
We would use an evaluation engine/library (there are many out there) to do the above, instead of the code that you have written to manually deal with those values. Does this make sense?

Reply:
He suggested so because he felt like we have found this too complicated to solve. But what i am suggesting with criteria evaluation is simple and more flexible because there are so many evaluations engines out there that we can just use. Does this make sense?

Reply:
Not really. You are always skeptical using external tools to solve basic problems. So I am okay changing the way criteria is structured but I am curious why not use regex. And if we have to use an external tool, kindly suggest one.

Reply:
The problem of evaluating mathematical expressions is such complicated that i would not recommend attempting to solve it. Besides, it would be reinventing the wheel because there are many tools out there which do exactly that.
We could for instance use velocity expressions or a JavaScript engine.

Reply:
@dicksonmulli here is an example of what i am talking about: https://pastebin.com/5u3JpdTj
From this code snippet, you can see that the role of the criteria is to simply evaluate to either `true` or `false`.
And with this approach, we can define a set of Java functions that we pass to the velocity context to get patient observations for things like Renal function, Pregnancy status, Weight or BMI, etc. Then some one can construct  expressions using any of them or whatever combination they want. We do not have to deal with the parsing of expressions or complexity of evaluating criteria like: `(($patient.getAge() > 5 && '$patient.getGender()' == 'M')  ||. ($patient.getAge() < 5 && '$patient.getGender()' == 'F'))`. Does this make sense?

Reply:
Yes, this makes sense. Thank you for explaining. I will make the changes and update PR.

Reply:
@dicksonmulli and FWIW, you can just use the velocity evaluator without the JavaScript engine by simply changing the criteria to this: `final String CRITERIA = "#set( $criteria = $patient.getAge() > 5 || $patient.getGender() == 'M' )$criteria";` 

Then the out put of `writer.toString()` will be either `true` or `false` as needed.

Reply:
@dkayiwa  I didn't see this in good time. I have already implemented with Js engine if that's okay. I have also included gender evaluation in the logic (even though we will use this later).

Reply:
The Js engine code as i wrote it will not work for Java 17. So let us just use velocity alone.